src/main/java/cz/mzk/fedora/solrcorrection/SolrVcRecords.java

```diff
@@ -0,0 +1,68 @@
+package cz.mzk.fedora.solrcorrection;
+
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrInputDocument;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class SolrVcRecords {
+
+    private final SolrClient solrClient;
+    private final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+
+    public SolrVcRecords(SolrClient solrClient) {
+        this.solrClient = solrClient;
+    }
+
+    public void removeVc(String idVc, String uuid) throws SolrServerException, IOException {
+        SolrQuery query = createSolrQuery(uuid);
+        SolrClientApi.singleRequestAndApply(query, solrClient, createUnlinkConsumer(idVc));
+    }
+
+    @SuppressWarnings("unchecked")
+    private Consumer<SolrDocument> createUnlinkConsumer(String idVc) {
+        return solrDoc -> {
+            List<String> collections = (List<String>) solrDoc.getFieldValue("collection");
+
+            if ((collections != null) && (collections.contains(idVc))) {
+                collections.remove(idVc);
+                try {
+                    String uuid = (String) solrDoc.getFieldValue("PID");
+                    setVcFor(uuid, collections);
+                } catch (SolrServerException | IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        };
+    }
+
+    private SolrQuery createSolrQuery(String uuid) {
```

Comment:
`createSolrQuery` nerika co to bude za query, prilis obecny nazev, co treba `createQueryByPidPath`?